base/sort.jl

```diff
@@ -560,15 +560,52 @@ elements that are not
 @inline send_to_end!(f::F, v::AbstractVector, ::ReverseOrdering, end_stable=false; lo, hi) where F <: Function =
     end_stable ? (send_to_end!(!f, v; lo, hi)+1, hi) : (hi-send_to_end!(f, view(v, hi:-1:lo))+1, hi)
 
+"""
+    send_to_end_stable!(f::Function, v::AbstractVector; [lo, hi])
+
+Send every element of `v` for which `f` returns `true` to the end of the vector and return
+the index of the last element which for which `f` returns `false`.
+
+`send_to_end_stable!(f, v, lo, hi)` is equivalent to `send_to_end_stable!(f, view(v, lo:hi))+lo-1`
+
+Preserves the order of the elements.
+"""
+function send_to_end_stable!(f::F, v::AbstractVector; lo=firstindex(v), hi=lastindex(v)) where F <: Function
+    tmp=copy(v)
```

Comment:
Ideally, this should utilize the scratch space re-use system, though it this PR is a performance improvement without, then that isn't essential.

Reply:
I've tried to re-use the scratch array, but not sure if i got your `@getkw` concept right.